backends/bmv2/psa_switch/psaSwitch.cpp

```diff
@@ -684,6 +684,41 @@ void ExternConverter_Counter::convertExternInstance(
     jctr->emplace("size", sz->to<IR::Constant>()->value);
     jctr->emplace("is_direct", false);
     ctxt->json->counters->append(jctr);
+
+    // Code below used to add json into EXTERN_INSTANCES NODE
+
+    auto extern_obj = new Util::JsonObject();
+    extern_obj->emplace("name", name);
+    extern_obj->emplace("id", nextId("extern_instances"));
+    extern_obj->emplace("source_info", inst->sourceInfoJsonObj());
+    ctxt->json->externs->append(extern_obj);
+    Util::JsonArray *arr = ctxt->json->insert_array_field(extern_obj, "attributes");
+
+    // first argument to create a counter is just a number, convert and dump to json
+    // we get a name from param, type and value from the arguments
+    auto attr_obj = new Util::JsonObject();
+    auto arg1 = sz->to<IR::Constant>();
+    auto param1 = eb->getConstructorParameters()->getParameter(0);
```

Comment:
even here you may want to check whether there exists a parameter 0. if someone compiles the wrong model file (or edits the model file) there won't be a parameter 0 and this will crash.

Reply:
Changed

Reply:
there's no need to reply to every comment, git will close some of them automatically